### PR TITLE
Gzip support

### PR DESCRIPTION
**Pull request by idohalevi**

### CODE REVIEW

File: lib/logzio-nodejs.js [hunk 1]
```diff
@@ -158,17 +159,19 @@ LogzioLogger.prototype._addTimestamp = function(msg) {
...
+        msg = {
+            message: msg
+        };
     }
     msg = _assign(msg, this.extraFields);
-    msg.type = this.type;
-
+    if (!msg.type) {
+        msg.type = this.type;
+    }
```
Comment (idohalevi) [inline lib/logzio-nodejs.js:174]: add in docs

File: README.md [hunk 2]
```diff
@@ -45,6 +45,7 @@ logger.log(obj);
 * **callback** - A callback function called when an unrecoverable error has occured in the logger. The function API is: function(err) - err being the Error object.
 * **timeout** - The read/write/connection timeout in milliseconds.
 * **addTimestampWithNanoSecs** - Add a timestamp with nano seconds granularity. This is needed when many logs are sent in the same millisecond, so you can properly order the logs in kibana. The added timestamp field will be `@timestamp_nano` Default: `false`
+* **doCompress** - If true the the logs are compressed in gzip format. Default: `false`
```
Comment (AmirTugi) [inline README.md]: No need for the `do` verb. `compress` is enough

File: test/test-logger.js [hunk 3]
```diff
@@ -103,14 +112,66 @@ describe('logger', function () {
...
+                message: 'hello there from test'
+            };
+
+            logger.log(logMsg);
+
+            function onDone() {
+                assert(logger._tryToSend.getCall(0).args[0].headers['content-encoding'] == 'gzip');
+                var unzipBody = JSON.parse(zlib.gunzipSync(logger._tryToSend.getCall(0).args[0].body));
+                assert(unzipBody.message == logMsg.message);
+                assert(unzipBody.extraField1 == 'val1');
```
Comment (AmirTugi) [inline test/test-logger.js]: These 2 lines are making a duplication of the value.
Use the `logger` object to read the value you want to check.

**Review by AmirTugi — APPROVED** [review]
Looks good

**Pull request merged**